clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -128,15 +136,39 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 				return c.CoreV1().Pods(p.selector.Namespace).Watch(context.TODO(), options)
 			},
 		},
-		p.checkPod,
+		p.addEvent,
 	)
+	go p.processEvents()
 	return informer.StartAndSync(i, p.stopCh, informerSyncTimeout)
 }
 
+func (p *podStartupLatencyMeasurement) addEvent(_, obj interface{}) {
+	event := &eventData{obj: obj, recvTime: time.Now()}
+	p.eventQueue.Add(event)
+}
+
+func (p *podStartupLatencyMeasurement) processEvents() {
+	for {
+		item, quit := p.eventQueue.Get()
+		if quit {
+			return
+		}
+
+		event, ok := item.(*eventData)
+		if !ok {
+			p.eventQueue.Done(item)
```

Comment:
continue? error?

Reply:
Actually - can you instead use this pattern:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L518-L538

?

Reply:
Done.